### PR TITLE
[Rust FN] Adding max string to module query

### PR DESCRIPTION
**Pull request by healthydeve**
## Description 

Originally the key, value for rocks db pairing are indexed in such a way that they are lexographically organized.
(i.e:
 (package, module, function)
 (abc, abc, abc), (abc, bcd, abc), (abc, cde, abc), (abc, cde, bcd)
).

this means when we pass in a null string, to find a key like so: (abc, '','') it will skip to the end of the DB, and show that there are no entries at all. This type of indexing caused issues in pagination whenever we pass in reverse = True because it will first skip to a different place in the DB. In this PR, we address those underlying issues and ensure that the DB will iterate backwards properly when using the filter portion of the `MoveFunction` filter type.

## Test Plan 
Checked that we cannot publish modules / function with names too long locally. 
image

False with nulls: 
image

True with nulls:
image

image
Sui system behavior works as intended
---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-storage/src/indexes.rs [hunk 1]
```diff
@@ -705,12 +705,22 @@ impl IndexStore {
             TxSequenceNumber::MIN
         });
 
-        let key = (
-            package,
-            module.clone().unwrap_or_default(),
-            function.clone().unwrap_or_default(),
-            cursor_val,
-        );
+        const MAX_STRING: &str =
```
Comment (healthydeve) [inline crates/sui-storage/src/indexes.rs:708]: Where should we store this? 

Can also change this to load dynamically `max_type_nodes` value.

**Review by wlmyng — APPROVED** [review]

**Comment by oxade** [comment]
@healthydeve where is the max enforce on writing to the store?
Seems you're only enforcing on read

**Comment by healthydeve** [comment]
> @healthydeve where is the max enforce on writing to the store? Seems you're only enforcing on read
> 
> This approach doesnt stop someone from writing a large string to the store.

There definitely is an enforcement of this because I tried publishing a package name with super long string, and the move-compiler passes back an error

**Review by oxade — DISMISSED** [review]
Lets put a hold on this for a sec till we discuss and ensure all angles are covered.

**Pull request merged**